packages/sdk/src/conversation/interface.ts

```diff
@@ -163,6 +171,36 @@ export interface TeamsFxBotCommandHandler {
   ): Promise<string | Partial<Activity> | void>;
 }
 
+/**
+ * Interface for a command handler that can process sso command to a TeamsFx bot and return a response.
+ */
+export interface TeamsFxBotSsoCommandHandler {
+  /**
+   * command id used to create sso command dialog, if not assigned, it will generate random command id
+   */
+  commandId?: string;
+
+  /**
+   * The string or regular expression patterns that can trigger this handler.
+   */
+  triggerPatterns: TriggerPatterns;
+
+  /**
+   * Handles a bot command received activity.
+   *
+   * @param context The bot context.
+   * @param message The command message the user types from Teams.
+   * @param ssoToken The sso token which can be used to exchange access token for the bot.
+   * @returns A `Promise` representing an activity or text to send as the command response.
+   * Or no return value if developers want to send the response activity by themselves in this method.
+   */
+  handleCommandReceived(
+    context: TurnContext,
+    message: CommandMessage,
+    ssoToken: string
```

Comment:
Does this token string contains expiration info, or, how long does it ensure before expired?

Reply:
Yes, expiration time can be parsed directly from ssoToken as below:
```
const ssoTokenExpiration = parseJwt(ssoToken).exp;
```

Reply:
Is it possible to provide something like `tokenProvider` instead of the one-time token itself.
I believe `tokenProvider` brings more benefits:
- Totally let user to determine whether to call `getToken`, for better performance and avoid potential expire risk.
- Totally let user to determine the token scope.
- Even can merge `TeamsFxBotCommandHandler` and `TeamsFxBotSsoCommandHandler` into one interface, and save user's learning curve.
- ...

I'm not sure if there's any technical limitation, but from auth UX a `provider` is better than `token` itself. (That's also how MSAL works.)

Reply:
Or, if not having `provider`, how to avoid token expired error?
E.g., the handler logic takes more than 10 minutes and the input `ssoToken` is expired when used. Is there any limitation that `ssoToken` only ensures xxx minutes before expired, or any workaround to get a new token?

Reply:
We cannot give a tokenProvider for user to get sso token directly, because sso token is come from Teams Client, for botframework, it is backend service, and it needs to use dialog to ask user to login and then obtain sso token from teams client. User can use this sso token to exchange access token for Graph as below:

```
    const teamsfx = new TeamsFx().setSsoToken(ssoToken);
    const token =  teamsfx.getToken(["User.Read"])
```

You can compare with the teams js sdk, it just return a ssoToken, then you can use this ssoToken for OBO flow to exchange real access token (need user consent, botframwork put the consent step in dialog) :

```ts
 microsoftTeams.initialize(() => {
          microsoftTeams.authentication.getAuthToken({
            successCallback: (ssoToken: string) => {
              resolve(ssoToken);
            },
            failureCallback: (errMessage: string) => {
              const errorMsg = "Get SSO token failed with error: " + errMessage;
              internalLogger.error(errorMsg);
              reject(new ErrorWithCode(errorMsg, ErrorCode.InternalError));
            },
            resources: [],
          });
        });
```

Reply:
Make sense that bot needs to trigger dialog to get token from Teams Client, just curious is it possible to proactively trigger the dialog (with turnContext and ssoConfig as inputs).

BTW, teams js sdk is more like `provider` since user can explicitly call `token = await teamsJs.authentication.getAuthToken(...)` to get token.

Let me try to build a sample and get back to this.

Reply:
I tried provider-style and it works. I modified code based on the bot-sso sample, by just adding following code to `teamsBot.ts` L60.

``` typescript
      if (txt === "test123") {
        /** wrap this as `token provider` */
        let deferredToken: Deferred<string>;
        const tokenPromise: Promise<string> = new Promise<string>(
          (resolve, reject) => (deferredToken = { resolve, reject })
        );
        this.dialog.setSSOOperation(async (text, token) => {
          deferredToken.resolve(token);
        });
        await this.dialog.run(context, this.dialogState);
        /** *************************** */

        const token = await tokenPromise;
        await context.sendActivity(token);
      }
```

And here's the teams result (it supports consent as well):
<img width="470" alt="image" src="https://user-images.githubusercontent.com/7642967/187342977-86637056-2e5a-4ce8-a389-f04c6b551cc9.png">

Actually, what I mean for `tokenProvider` is to provide a way to **proactively trigger sso dialog**. There's also similar issue (https://stackoverflow.com/questions/56975804/how-to-send-a-proactive-dialog-with-bot-framework) with certain answers.

Anyway, it's up to you for the final design. My concern is that - if we're going to add SSO to proactively messaging (e.g., notification), `provider` is always needed.

Reply:
Thanks, I will investigate this

Reply:
Offline discussed and understand the distributed limitation...
Agree that currently hard to have the `provider`-style solution. THX~